wire-ios-sync-engine/Source/Synchronization/Strategies/PrekeyGenerator.swift

```diff
@@ -0,0 +1,65 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireCryptobox
+
+class PrekeyGenerator {
+
+    // This is needed to save ~3 seconds for every unit test run
+    // as generating 100 keys is an expensive operation
+    static var _test_overrideNumberOfKeys: UInt16?
+
+    let proteusProvider: ProteusProviding
+    let keyCount: UInt16 = _test_overrideNumberOfKeys ??  100
+
+    init(proteusProvider: ProteusProviding) {
+        self.proteusProvider = proteusProvider
+    }
```

Comment:
What about removing the static variable and just having another init argument?

```suggestion
import WireCryptobox

struct PrekeyGenerator {

    let proteusProvider: ProteusProviding
    let keyCount: UInt16

    init(proteusProvider: ProteusProviding, keyCount: UInt16 = 100) {
        self.proteusProvider = proteusProvider
        self.keyCount = keyCount
    }
```

Reply:
Yep that would be nicer, but I need to refactor the user session init then though so I can inject the value.

I'll see if it's doable without to much changes

Reply:
@caldrian Seeing the depth of dependencies, I'll add a note to it and tackle it when we review dependencies injection

SessionManager -> ZMUserSession -> StrategiesDirectory -> UserClientRequestStrategy -> PrekeyGenerator